### PR TITLE
Convert download code to kotlin

### PR DESCRIPTION
**Pull request by maskaravivek**

### CODE REVIEW

**Comment by macgills** [comment]
@maskaravivek should this have been merged without review?

**Comment by maskaravivek** [comment]
Apologies for merging it without a review. It was a minor change but if you have any comments then I would be happy to address it.

**Comment by macgills** [comment]
I'll leave a few now

File: app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @param m Media file to download
+     */
+    @JvmStatic
+    fun downloadMedia(activity: Activity?, m: Media) {
+        val imageUrl = m.getImageUrl()
+        var fileName = m.getFilename()
+        if (imageUrl == null || fileName == null || activity == null
+        ) {
+            Timber.d(
+                "Skipping download media as either imageUrl %s or filename %s activity is null",
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt:28]: We can use kotlin string interpolation here

File: app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                imageUrl, fileName
+            )
+            return
+        }
+        // Strip 'File:' from beginning of filename, we really shouldn't store it
+        fileName = fileName.replaceFirst("^File:".toRegex(), "")
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt:34]: `.substringAfter` will express this a bit nicer

File: app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        val imageUri = Uri.parse(imageUrl)
+        val req = DownloadManager.Request(imageUri)
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt:36]: we can use `.apply` to make building this Request look nicer

File: app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+object DownloadUtils {
+    /**
+     * Start the media file downloading to the local SD card/storage. The file can then be opened in
+     * Gallery or other apps.
+     *
+     * @param m Media file to download
+     */
+    @JvmStatic
+    fun downloadMedia(activity: Activity?, m: Media) {
+        val imageUrl = m.getImageUrl()
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt:23]: shouldn't the IDE be suggesting using propertyAccess syntax?

File: app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        var fileName = m.getFilename()
+        if (imageUrl == null || fileName == null || activity == null
+        ) {
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/utils/DownloadUtils.kt:26]: weird place for a parentheses

**Comment by maskaravivek** [comment]
All of them are just cosmetic changes but I have addressed them anyways. #3669.

**Pull request merged**